src/ir_passes/await.ml

```diff
@@ -324,29 +341,45 @@ and c_exp' context exp k =
                at = exp.at;
                note = Note.{ exp.note with typ = typ' } }))
     end)
-  | TryE (exp1, cases) ->
+  | TryE (exp1, cases, finally_opt) ->
+     let pre k =
+       match finally_opt with
+       | Some (id2, typ2) -> precont k (var id2 typ2)
+       | None -> k in
+    let pre' = function
+      | Cont k -> Cont (pre k)
+      | Label -> assert false in
+    (* All control-flow out must pass through the potential `finally` thunk *)
+    let context = LabelEnv.map pre' context in
+    (* assert that a surrounding `AwaitPrim _` has set up a `Cleanup` cont *)
```

Comment:
I don't grokk this comment

Reply:
Some `AwaitPrim` in the enclosing context is in charge of setting up a trivial-valued `Cleanup` key in the `LabelEnv`. We depend on it being there when mapping over the `context`.